docs/6.x/pack.md

```diff
@@ -805,6 +811,33 @@ correct image references.
     There is a sample application available on [GitHub](https://github.com/gravitational/quickstart/tree/master/mattermost)
     that demonstrates this workflow.
 
+### Customizing Helm values
+
+It is possible to customize values of your Helm charts when installing or
+upgrading the application. To provide custom Helm values at install time,
+pass them via `--values` and `--set` flags to `gravity install` command:
+
+```bash
+unpacked-image$ ./gravity install --values=custom-values.yaml --set=nginx.image=1.9.1 --set=postgres.registry=internal.registry.io
+```
+
+The provided values are merged into a single values file that is mounted
+into install and post-install hooks under `/var/lib/gravity/helm/values.yaml`
+so the install hook can use it in the `helm install` command:
+
+```yaml
+...
+command: ["/usr/local/bin/helm", "install", "/var/lib/gravity/resources/charts/example", "--values", "/var/lib/gravity/helm/values.yaml"]
+```
+
+The same is true for upgrades: both the `./upgrade` script included with the cluster
+image and `gravity upgrade` commands support providing custom Helm values which
+will get mounted at the same location in the upgrade/post-upgrade hooks:
+
+```bash
+unpacked-image$ ./upgrade --values=custom-values.yaml --set=nginx.image=1.11.0
```

Comment:
Does this imply that any customization / values needs to be saved and maintained for an upgrade and provided by the customer?

Reply:
If the user needs to customize some values during upgrade, they will need to provide those to the upgrade command, yes. The "regular" (or "default") values file can still be packaged together with their chart.

Reply:
Well there is that aspect, but the angle I was thinking of, is I customize a bunch of values when I install the chart. What happens later when I upgrade, does it keep the values file from install time, or do I as a user need to keep a copy and remember to use it again during an upgrade?

Reply:
@knisbet Good question - not necessarily, we don't need to save values customized during the install ourselves. These values are saved by Helm and can be retrieved for any release using `helm get values` command for any revision of the release. So the upgrade hook can just take these into account if needed.

What this makes me think about though, is that we should probably save Helm values provided during `tele build`, because these are not available anywhere else. So if let's say you override an image during tele build but forget to override it during install, we may end up attempting to pull it from the internet cause different version was vendored.

I'll address this separately after this one (I'm thinking to either keep a separate values file with values set during tele build and mount it inside hooks as well, or just merge it into the same values.yaml that this PR adds).